src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,173 @@
+import React, {Component} from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import {redirectToLastReport} from '../../libs/actions/App';
+import IOUAmountPage from './steps/IOUAmountPage';
+import IOUParticipantsPage from './steps/IOUParticipantsPage';
+import IOUConfirmPage from './steps/IOUConfirmPage';
+import Header from '../../components/Header';
+import styles from '../../styles/styles';
+import Icon from '../../components/Icon';
+import getPreferredCurrency from '../../libs/actions/IOU';
+import {Close, BackArrow} from '../../components/Icon/Expensicons';
+
+/**
+ * IOU modal for requesting money and splitting bills.
+ */
+const propTypes = {
+
+    /* Onyx Props */
+    // Url currently in view
+    currentURL: PropTypes.string,
+
+    // IOU modal data
+    iouData: PropTypes.shape({
+        isLoadingCurrency: PropTypes.bool.isRequired,
+        isLoadingParticipants: PropTypes.bool.isRequired,
+        createReportInProgress: PropTypes.bool.isRequired,
+    }),
```

Comment:
CC @marcaaron / @tgolen 

I feel like this might not be the best Onyx key naming convention. I tried setting up sub-objects for each step, but this became quite verbose (especially the PropType declaration).  Any thoughts or suggestions?

Also note these aren't all used yet, but are left as a building block for the external contributors.

Reply:
I think one thing that could clean this up is to just use a single `IOU.IS_LOADING` key, that could be applied to any step. Then there wouldn't be a need for three different ones.

Reply:
I've made the change and it seems fine in this case, but I can see potential issues in the future if two closely linked Components depend on the same Onyx key.